clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/NoInvalidationServerStoreProxy.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.client.internal.store;
+
+import org.ehcache.clustered.client.internal.EhcacheClientEntity;
+import org.ehcache.clustered.client.internal.EhcacheEntityOperationException;
+import org.ehcache.clustered.common.messages.EhcacheEntityResponse;
+import org.ehcache.clustered.common.messages.ServerStoreMessageFactory;
+import org.ehcache.clustered.common.store.Chain;
+
+import java.nio.ByteBuffer;
+
+/**
+ * Provides client-side access to the services of a {@code ServerStore}.
+ */
+class NoInvalidationServerStoreProxy implements ServerStoreProxy {
+
+  private final ServerStoreMessageFactory messageFactory;
+  private final EhcacheClientEntity entity;
+
+  NoInvalidationServerStoreProxy(ServerStoreMessageFactory messageFactory, final EhcacheClientEntity entity) {
+    this.messageFactory = messageFactory;
+    this.entity = entity;
+  }
+
+  @Override
+  public String getCacheId() {
+    return messageFactory.getCacheId();
+  }
+
+  @Override
+  public void addInvalidationListener(InvalidationListener listener) {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public Chain get(long key) {
+    EhcacheEntityResponse response;
+    try {
+      response = entity.invoke(messageFactory.getOperation(key), false);
+    } catch (Exception e) {
+      throw new ServerStoreProxyException(e);
+    }
+    if (response != null && response.getType() == EhcacheEntityResponse.Type.GET_RESPONSE) {
+      return ((EhcacheEntityResponse.GetResponse)response).getChain();
+    } else {
+      throw new ServerStoreProxyException("Response for get operation was invalid : " +
+                                          response != null ? response.getType().toString() : "null message");
```

Comment:
That's wrong - was fixed in master in the ServerStoreProxy and so fix needs to be applied here.